src/main/java/org/pac4j/vertx/handler/impl/RequiresAuthenticationHandler.java

```diff
@@ -138,6 +145,33 @@ public void handle(RoutingContext routingContext) {
 
     }
 
+    @Override
+    protected void authorise(final User user, final RoutingContext context) {
+        if (! (user instanceof Pac4jUser)) {
+            throw new TechnicalException("Wrong user type in authorise");
+        }
+
+        Handler<AsyncResult<Boolean>> authHandler = res -> {
+            if (res.succeeded()) {
+                if (res.result()) {
+                    context.next();
+                } else {
+                    context.fail(403);
+                }
+            } else {
+                context.fail(res.cause());
```

Comment:
Sorry for the unclear question. I was wondering about the status code (500 as you answered).